### ISSUE TITLE
Invalid flag goes out of sync when text field is inside custom field

### ISSUE THREAD

**Issue by vaadin-miki** [vaadin/flow-components#1076]
### Description
Note: this all has been done using Java. I have no idea if it is reproducable with pure client-side code, but probably is.

When text field is placed inside a custom field, something overrides `invalid` flag to be false, despite server-side code setting it to true. The text field has a pattern that "guards" the input, maybe it is related.

### Expected outcome
The invalid flag stays in sync with whatever has been set by the server.

### Actual outcome
The flag is set to `false` making the component not display validation error.

### Live Demo and steps to reproduce
This issue was spotted in my add-on here: https://github.com/vaadin-miki/super-fields/issues/241
The thread includes code that reproduces the issue. In my server-side code my custom field has this code:
```
setInvalid:
        super.setInvalid(invalid);
        this.field.setInvalid(invalid);
```
In certain cases the `invalid` flag on the custom field is `true`, while the same flag on the text field is `false`, when obviously both should be the same.

I have caught the property change in the client-side and here is the relevant log output:
```
org.vaadin.miki.BinderPage - value changed ComponentValueChangeEvent[source=org.vaadin.miki.superfields.numbers.SuperBigDecimalField@7a4ac994, value = 12345, oldValue = 0]
org.vaadin.miki.BinderPage - validator validating 12345
org.vaadin.miki.superfields.numbers.SuperBigDecimalField - setting invalid to false
org.vaadin.miki.superfields.numbers.SuperBigDecimalField - setting invalid to true
org.vaadin.miki.superfields.numbers.SuperBigDecimalField - client-side property changed to true
org.vaadin.miki.superfields.numbers.SuperBigDecimalField - client-side property changed to false
```

There is more discussion about a similar behaviour here: https://github.com/vaadin/vaadin-date-picker/issues/751 - while I have not tested it, I suspect it is caused by the same issue (invalid flag being overwritten by client-side code).

### Server-side fix
Adding this code to custom field's `setInvalid` seems to fix the issue:
```
        this.field.getElement().getNode().runWhenAttached(ui -> ui.beforeClientResponse(this.field, context ->
                this.field.getElement().setProperty("invalid", super.isInvalid())
        ));
```
However, it simply sets the flag again and thus degrades performance. I think the server-indicated validation should simply not be overridden by anything.

(please note that this will work only because the custom field has only one component, the text field - this may not be a good solution when there are more fields)

### Browsers Affected
All. Present in Vaadin 14.4.

**Comment by vursen** [comment]
Hello @vaadin-miki!

Could you please check if #3277 has solved the problem? Otherwise, could you provide a more simple reproduction with a minimal number of steps? There are currently too many things mixed in one example so I couldn't really identify the actual problem.

**Comment by vursen** [comment]
Assuming that the described issue has been fixed as part of #3277, I'm closing the ticket. Feel free to re-open the ticket if the issue still persists.
